internal/pam_issuer_server/handler.go

```diff
@@ -0,0 +1,389 @@
+//go:build linux
+
+package pam_issuer_server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	pamapi "github.com/flightctl/flightctl/api/v1alpha1/pam-issuer"
+	"github.com/flightctl/flightctl/internal/auth/issuer/pam"
+	"github.com/flightctl/flightctl/internal/config"
+	"github.com/flightctl/flightctl/internal/crypto"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+)
+
+// Handler implements the PAM issuer API handlers
+type Handler struct {
+	log         logrus.FieldLogger
+	cfg         *config.Config
+	pamProvider *pam.PAMOIDCProvider
+	cancel      context.CancelFunc
+	wg          sync.WaitGroup
+}
+
+// NewHandler creates a new PAM issuer handler
+func NewHandler(
+	log logrus.FieldLogger,
+	cfg *config.Config,
+	ca *crypto.CAClient,
+) (*Handler, error) {
+	if cfg.Auth == nil || cfg.Auth.PAMOIDCIssuer == nil {
+		return nil, fmt.Errorf("PAM OIDC issuer not configured")
+	}
+
+	pamProvider, err := pam.NewPAMOIDCProvider(ca, cfg.Auth.PAMOIDCIssuer)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create PAM OIDC provider: %w", err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
the ctx does not escape the constructor does it? generally speaking it is unexpected to have a ctx created or consumed in a constructor.